### DISCUSSION TITLE
Will parcels calculate w?

### DISCUSSION THREAD

**Discussion by jklymak** [Parcels-code/Parcels#1133] [Q&A] [answered]
I have a NEMO run where W was not saved.  Will parcels infer a vertical velocity for me, or do I need to calculate one myself?  Thanks!

**Comment by erikvansebille** [comment, marked as answer]
Hi @jklymak. No, Parcels does not have any methods to calculate `W` on-the-fly. You will have to do that yourself, I'm afraid

**Reply by jklymak** [reply]
Thanks.  Is it worth putting in an issue suggesting to add that?  Unlike the naive user, Parcels has the grid metrics readily available, and I think its relatively common for folks to not save diagnostic variables like W for hydrostatic simulations.

**Reply by erikvansebille** [reply]
No, this is not within scope for Parcels. Calculating W is very sensitive to the exact formulations of the advection-equation, and therefore is different for each OGCM. We will never be able to cover this for all OGCMs, and therefore don't even try. 

Most models do have codes available to calculate W from stored fields, you should run these yourself and can then use the output as input in Parcels
